v2/context/context.go

```diff
@@ -50,3 +51,29 @@ func TopicFrom(ctx context.Context) string {
 	}
 	return ""
 }
+
+// Opaque key type used to store linear backoff parameters
+type linearBackoffKeyType struct{}
+
+var linearBackoffKey = linearBackoffKeyType{}
+
+// WithLinearBackoff returns back a new context with the linear backoff parameters.
+// MaxTries is the maximum number for retries and delay is the time interval between retries
+func WithLinearBackoff(ctx context.Context, delay time.Duration, maxTries int) context.Context {
+	return context.WithValue(ctx, linearBackoffKey, Backoff{
+		Strategy: BackoffStrategyLinear,
```

Comment:
Exactly. The goal here is if we add a new strategy, it will override any previously set strategies. 

So down the road when we make ExpBackoff we can make a method called `WithExponentialBackoff` and all the plumbing is done.

The aim with `WithLinearBackoff` is to keep the consumer api simple and they do not have to be aware of the details of the Backoff Struct

Reply:
would `WithRetiesLinearBackoff` make more sense?

Reply:
done.